course-v2/assets/course-v2.ts

```diff
@@ -26,13 +32,20 @@ $(function() {
   initCourseDrawersClosingViaSwiping()
 })
 
-// @ts-expect-error for window.initVideoJS()
+let videoJSLoaded = false
+let nanogallery2Loaded = false
+
 window.initVideoJS = () => {
+  if (videoJSLoaded) return
   initVideoDownloadPopup()
   import("./videojs-imports").then(module => {
     module.initVideoJS()
   })
+  videoJSLoaded = true
 }
 
-// @ts-expect-error for window.initNanogallery2()
-window.initNanogallery2 = () => import("./nanogallery2-imports")
+window.initNanogallery2 = () => {
+  if (nanogallery2Loaded) return
+  import("./nanogallery2-imports")
```

Comment:
@ibrahimjaved12 By the way: I expect that this import was not actually problematic, but it doesn't hurt to make it happen only once.

Suppose:
```ts
//somefile.js
console.log("Hello from somefile.js")

// main file:

import("./somefile").then(() => {
    console.log('Somefile has loaded, A')
})

import("./somefile").then(() => {
    console.log('Somefile has loaded, B')
})
```
The logs should be:
```

```
Hello from somefile.js
Somefile has loaded, A // The import promise has resolved, so the `then` runes
Somefile has loaded, B // The import promise has resolved (but does not re-run the imported code)
```

When you import a file, its contents only get run once.